LoopKit/InsulinKit/InsulinDeliveryStore.swift

```diff
@@ -375,6 +401,27 @@ extension InsulinDeliveryStore {
         return doses
     }
 
+    /// Deletes objects from the cache that match the given sample UUID
+    ///
+    /// - Parameter uuid: The UUID of the sample to delete
+    /// - Returns: Whether the deletion was made
+    private func deleteCachedObjects(forSampleUUIDs uuids: [UUID], batchSize: Int = 500) -> Int {
+        dispatchPrecondition(condition: .onQueue(queue))
+
+        var deleted = 0
+
+        cacheStore.managedObjectContext.performAndWait {
+
+            for batch in uuids.chunked(into: batchSize) {
+                let predicate = NSPredicate(format: "uuid IN %@", batch.map { $0 as NSUUID })
+                if let count = try? cacheStore.managedObjectContext.purgeObjects(of: CachedInsulinDeliveryObject.self, matching: predicate) {
```

Comment:
[nit] If this fails (not sure why it would, but...) then the local cache will not ever match HealthKit (until the non-deleted samples expire).

Reply:
Updating the set of calls to not advance query anchor on failures to delete or insert.